dask/multiprocessing.py

```diff
@@ -21,11 +20,25 @@ def _reduce_method_descriptor(m):
 copyreg.pickle(type(set.union), _reduce_method_descriptor)
 
 
-def _dumps(x):
-    return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+try:
+    import cloudpickle
+
+    _dumps = partial(cloudpickle.dumps, protocol=pickle.HIGHEST_PROTOCOL)
+    _loads = cloudpickle.loads
+except ImportError:
 
+    def _dumps(obj):
+        try:
+            return pickle.dumps(obj, protocol=pickle.HIGHEST_PROTOCOL)
+        except (pickle.PicklingError, AttributeError):
+            warn(
+                "pickle failed; install cloudpickle to allow more objects "
+                "(such as lambdas and locally defined functions) to be pickled",
+                UserWarning,
```

Comment:
This warning will probably not be well understood by most of our users, who don't know what pickle or lambdas are.  The previous error message, *"Please install cloudpickle to use the multiprocessing scheduler"* was, I think, more clear to people and didn't seem to cause any problems (no one complained about this). 

I think I'm personally inclined to fail early if cloudpickle isn't installed, rather than fail at serialization time.  I think that this is likely to reduce confusion among novice users in general, even if we reduce functionality slightly.  What do you think?

Reply:
I'm removing the parenthesis

Reply:
I still think that this is an issue.  Most users don't understand what pickle is, or even what serialization is.  I think that the previous error message here was more clear and less prone to put novice users into a sitution where they didn't know how to resolve their problem.

```python
        elif scheduler.lower() in ["processes", "multiprocessing"]:	
            raise ValueError(	
                "Please install cloudpickle to use the '%s' scheduler." % scheduler	
            )	
```

Reply:
How about now?